### PR TITLE
CLI `compile` docs

### PR DESCRIPTION
**Pull request by dwhswenson**
Docs for features in CLI 0.3.0 were missing from the main documentation. (I forgot open this PR, and honestly, I'm impressed with some of the nice stuff I put in here -- local docs generation catches things in my local environment that haven't been published yet: and even that looks good without any real effort on this.)

This also includes some ... easter egg? docs for the real "wizards" out there.

### CODE REVIEW

**Comment by dwhswenson** [comment]
This is ready for review and comment. Without review, I will merge this after at least 24 hours, no sooner than Sat 28 Sep 03:00 GMT (Fri 27 Sep 22:00 my local).

I'd suggest trying this out on you own machine. We'll only see failures *after* merge, since we're not currently testing docs builds in PRs (thing to change).

File: docs/cli/wizard/index.rst
Comment (sroet) [inline docs/cli/wizard/index.rst:103]: Could this not be checked in the answer sanitation of the wizard? (just to be **extra** friendly) (not needed for this PR, but might be nice to spin into its own issue)

File: docs/conf.py
Comment (sroet) [inline docs/conf.py:49]: shouldn't this directory be relative to the current file instead of the cwd?

**Review by sroet — APPROVED** [review]
Couple comments/questions. Feel free to ignore, any/all

Most critical one was the `python -m pip` one. LGTM otherwise

File: docs/cli/compile/input/categories.yml
Comment (dwhswenson) [reply docs/cli/compile/input/categories.yml]: Done in [5151c1a](https://github.com/openpathsampling/openpathsampling/pull/1159/commits/5151c1ab072373e845f911d3ecac22e88e8b013f)

File: docs/conf.py
Comment (dwhswenson) [reply docs/conf.py]: Done in [b9e2818](https://github.com/openpathsampling/openpathsampling/pull/1159/commits/b9e281856d3dcd9758bffcf902fcecc44480f0d7)

File: docs/cli/compile/file_format.rst
Comment (dwhswenson) [reply docs/cli/compile/file_format.rst:40]: It does locally; I think it (like all of this) needs the CLI to be installed to work, though.

File: docs/cli/compile/file_format.rst
Comment (dwhswenson) [reply docs/cli/compile/file_format.rst:54]: Probably? But I'm going to be lazy and skip that documentation here.

File: docs/cli/wizard/index.rst
Comment (dwhswenson) [reply docs/cli/wizard/index.rst:103]: Eh, going to leave this as is...  I'm not sure how much the Wizard will ever matter. It came out of a weird time late in pandemic lockdowns. I still find a few things about it hilarious, but I wouldn't honestly recommend it as an interface right now.

This documentation shouldn't even show up when a user browses the docs; they would need to know to look at the right URL.

**Pull request merged**
